### PR TITLE
Jseb

### PR DESCRIPTION
**Pull request by jeangit**
everything should be reverted now. (basically, adding `find_missing_examples.nim` again, and .gitignore…)
for .gitignore, i think it should stay local in a shared project like this.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -26,3 +26,7 @@ a.out
 /outputGotten.txt
 /testresults
 /tests/emscripten/*.js
+
+.gitignore
+*.elf
+*jseb*
```
Comment (greenfork) [inline .gitignore:32]: .gitignore is a file used by git to tell which files shouldn't be committed to the repository. Usually it is build artifacts or test artifacts. It should be for general use by everybody. So
- `.gitignore` file shouldn't be ignored
- `*jseb*` is strictly personal use, you should manage any personal files/folders by yourself

File: examples/models/models_first_person_maze.nim [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  initWindow(screenWidth, screenHeight, "raylib [models] example - first person maze")
+
+  var camera = Camera3D(
+    position: (0.2, 0.4, 0.2), target: (0.0, 0.0, 0.0), up: (0.0, 1.0, 0.0), fovy: 45.0, projection: Perspective
+  )
+  camera.setCameraMode(FIRST_PERSON)
+
+  let imMap:Image = loadImage("resources/cubicmap.png") # load cubicmap image (RAM)
+  let cubicMapTexture:Texture2D = loadTextureFromImage(imMap)  # convert image to texture (VRAM)
+  let mesh = genMeshCubicmap(imMap, ( x:1.0, y:1.0, z:1.0))
```
Comment (greenfork) [inline examples/models/models_first_person_maze.nim:29]: Let's use consistent style. Space between `:` and everything else, space between `(` and everything else. This is a good example of what style to use:
```nim
let playerPos:Vector2 = Vector2(x: camera.position.x, y: camera.position.z)
```

There are a lot of inconsistencies

File: examples/models/models_first_person_maze.nim [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    elif playerCellX >= cubicMapTexture.width: playerCellX = cubicMapTexture.width - 1
+    if playerCellY < 0: playerCellY = 0
+    elif playerCellY >= cubicMapTexture.height: playerCellY = cubicMapTexture.height - 1
+    # Check map collisions using image data and player position
+    # TODO: Improvement: Just check player surrounding cells for collision
+    for y in 0..<cubicMapTexture.height:
+      for x in 0..<cubicMapTexture.width:
+        var collisionRect = Rectangle(x: modelPosition.x - 0.5 + x.float, y: modelPosition.z - 0.5 + y.float, width: 1.0, height: 1.0)
+        # collision: white pixel, red channel
+        if mapPixels[y*cubicMapTexture.width + x].r == 255 and checkCollisionCircleRec(playerPos, playerRadius, collisionRect) == true:
```
Comment (greenfork) [inline examples/models/models_first_person_maze.nim:67]:
```suggestion
        if mapPixels[y*cubicMapTexture.width + x].r == 255 and checkCollisionCircleRec(playerPos, playerRadius, collisionRect):
```

**Review by greenfork — COMMENTED** [review]
Could you also follow steps 5 and 6 from converting examples? https://github.com/greenfork/nimraylib_now/blob/master/CONTRIBUTING.md

File: .gitignore (hunk 1 above)
Comment (jeangit) [reply .gitignore:32]: i will revert and put my personal configuration in another file (i see i could use ~/.gitignore for example)

File: examples/models/models_first_person_maze.nim (hunk 2 above)
Comment (jeangit) [reply examples/models/models_first_person_maze.nim:29]: ok for space after tuple/object name, but are your sure for a space between `(` and the rest ? I see i have forgotten to remove the one here as you had asked me  last day (but it was for function calling, so may be different here ?)

File: examples/models/models_first_person_maze.nim [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+          cubicMapTexture.width*4, cubicMapTexture.height*4, Green)
+      # Draw player position radar
+      drawRectangle(getScreenWidth() - cubicMapTexture.width*4 - 20 +
+          playerCellX*4, 20 + playerCellY*4, 4, 4, Yellow)
+      drawFPS(10, 10)
+
+  unloadImageColors(mapPixels[0].addr) # Unload color array
+  unloadTexture(cubicMapTexture) # Unload cubicmap texture
+  unloadTexture(mapTexture) # Unload map texture
+  unloadModel(model); # Unload map model
```
Comment (greenfork) [inline examples/models/models_first_person_maze.nim:90]:
```suggestion
  unloadModel(model) # Unload map model
```
I assume it's left from C

File: examples/models/models_first_person_maze.nim (hunk 2 above)
Comment (greenfork) [reply examples/models/models_first_person_maze.nim]: I see you used nimpretty, it generally should solve the problem of how to style it correctly. But sometimes it does strange indentation that should be fixed manually. In this case I think it's good enough

**Comment by greenfork** [comment]
> Could you also follow steps 5 and 6 from converting examples? https://github.com/greenfork/nimraylib_now/blob/master/CONTRIBUTING.md

I think you missed this comment?

File: examples/models/models_first_person_maze.nim (hunk 4 above)
Comment (jeangit) [reply examples/models/models_first_person_maze.nim:90]: yes i take the C source ans translate it gradually by chunks. As `;` is valid in Nim as separator, i've missed it.

**Comment by jeangit** [comment]
i have read CONTRIBUTING but was focalized of the source itself with NEP1 rules , and forgot the test suite part.

**Comment by greenfork** [comment]
@jeangit do you want to do final fixes or I can do them for you?

**Comment by jeangit** [comment]
sorry i haven't seen this message. Yes, you can tune as you want the sample.

**Comment by greenfork** [comment]
@jeangit no problem. And by the way, git saves the history of the code. For example, if you did some changes, it is recorded in the git history. And if I do any changes on top of it, the authorship of the code according to git will belong to us 2. So if you don't mind, I will just do several slight changes in this same PR and commit our work together. I think it is important to give the credit for the work like that.

**Pull request closed without merging**